js/src/sdk/models/connectedAccounts.ts

```diff
@@ -70,16 +79,31 @@ export class ConnectedAccounts {
 
     async initiate(payload: InitiateConnectionDataReq): Promise<ConnectionRequest> {
         try {
-            const {integrationId, entityId = 'default', labels,data={}, redirectUri} = payload;
-      
-            const res = await client.connections.initiateConnection({ body: {
-                integrationId,
-                entityId,
-                labels,
-                redirectUri,
-                data,
-            }  }).then(res => res.data);
-            
+            let { integrationId, entityId = 'default', labels, data = {}, redirectUri, authMode = '', authConfig = {}, appName } = payload;
```

Comment:
The initiate method should validate required parameters before proceeding. Consider adding input validation:
```typescript
if (!integrationId && !appName) {
    throw new Error('Either integrationId or appName must be provided');
}
if (!integrationId && !authMode) {
    throw new Error('authMode is required when creating new integration');
}
```